contracts/eosio.system/src/voting.cpp

```diff
@@ -65,6 +90,45 @@ namespace eosiosystem {
 
    }
 
+   void system_contract::regproducer( const name& producer, const eosio::public_key& producer_key, const std::string& url, uint16_t location ) {
+      require_auth( producer );
+      check( url.size() < 512, "url too long" );
+
+      check( producer_key.index() < 2, "currently only K1 and R1 producer keys are supported" );
+      check( !is_null_key( producer_key ), "public key should not be the default value" );
+
+      register_producer( producer, convert_to_block_signing_authority( producer_key ), url, location );
+   }
+
+   void system_contract::regproducer2( const name& producer, const eosio::block_signing_authority& producer_authority, const std::string& url, uint16_t location ) {
+      require_auth( producer );
+      check( url.size() < 512, "url too long" );
+
+      std::set<eosio::public_key> unique_keys;
+      std::visit( [&](auto&& auth ) {
+         uint32_t sum_weights = 0;
+
+         for (const auto& kw: auth.keys ) {
+            check( kw.key.index() < 2, "currently only K1 and R1 producer keys are supported" );
```

Comment:
I'm curious why we've gone to lengths to not support WebAuthn keys here.  If there is a technical reason then we should keep it but if it is just best-practices we should document it and consider not tightly coupling this application level code with a low-level detail like whether the key abstraction represents a specific set of implementations

Reply:
WebAuthn keys are allowed in the authorities.